### PR TITLE
tweak the log level for IOExceptions and add recommendation information to existing logging.

### PR DESCRIPTION
**Pull request by jkwatson**
I think this could help address issues with users not understand what's going on wrt. retries and network timeouts.

### CODE REVIEW

**Comment by breedx-nr** [comment]
Needs a rebase.

File: telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java [hunk 1]
```diff
@@ -188,7 +191,7 @@ private Response handle429(HttpResponse response, String responseBody)
     List<String> retryAfter = findHeader(responseHeaders, "retry-after");
     int retryAfterSeconds = getRetryAfterValue(response, responseBody, retryAfter);
     logger.warn(
-        "Response from New Relic ingest API: code: {}, body: {}, retry-after: {}",
+        "Response from New Relic ingest API. Retry with wait recommended.b: code: {}, body: {}, retry-after: {}",
```
Comment (breedx-nr) [inline telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java]: lil' typo

Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java]:
```suggestion
        "Response from New Relic ingest API. Retry with wait recommended : code: {}, body: {}, retry-after: {}",
```

Comment (jkwatson) [reply telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java]: fixed

**Pull request merged**